test/unit/rfc/paths_rule.cpp

```diff
@@ -190,13 +192,68 @@ class paths_rule_test
         bad<T>("/%25%2e");
     }
 
+    void
+    test_normalize()
```

Comment:
This should be broken up into individual functions like `testScheme`, `testAuthority`, etc. And then maybe one `testAll` at the end.

Reply:
There's no `remove_dot_segments` for scheme, authority, etc... Only paths. This function could be called `test_remove_dot_segments` but I was trying to type less.

This is just testing `remove_dot_segments`, which is an auxiliary algorithm that only applies to normalizing paths. The tests considering all components are in `url_view.cpp`.

Reply:
oh... well, `testPath` then.

Reply:
This is not very descriptive because all tests in this file are testing paths too

https://github.com/alandefreitas/url/blob/11c76e7395821e254d4bd0fa0edc8ccd0ea38a43/test/unit/rfc/paths_rule.cpp#L252-L256

Reply:
Thats a different file though. there's no need to pick globally unique names of test functions...